File: api/src/Entity/TypeRelation.php
<?php

namespace App\Entity;

use App\Repository\TypeRelationRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: TypeRelationRepository::class)]
class TypeRelation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 50)]
    private ?string $libelle = null;

    #[ORM\ManyToMany(targetEntity: Ressource::class, inversedBy: 'typeRelations')]
    private Collection $Ressource;

    public function __construct()
    {
        $this->Ressource = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getLibelle(): ?string
    {
        return $this->libelle;
    }

    public function setLibelle(string $libelle): static
    {
        $this->libelle = $libelle;

        return $this;
    }

    /**
     * @return Collection<int, Ressource>
     */
    public function getRessource(): Collection
    {
        return $this->Ressource;
    }

    public function addRessource(Ressource $ressource): static
    {
        if (!$this->Ressource->contains($ressource)) {
            $this->Ressource->add($ressource);
        }

        return $this;
    }

    public function removeRessource(Ressource $ressource): static
    {
        $this->Ressource->removeElement($ressource);

        return $this;
    }

    public function toArray()
    {
        return [
            'id' => $this->id,
            'name' => $this->libelle
        ];
    }
}

File: mobile/ressources_re_mobile/lib/classes/TypeRelation.dart
// ignore_for_file: file_names

import 'package:ressources_re_mobile/classes/Ressource.dart';

class TypeRelation {
  int? id;
  String? libelle;
  List<Ressource> ressources = [];

  TypeRelation();

  int? getId() {
    return id;
  }

  String? getLibelle() {
    return libelle;
  }

  void setLibelle(String libelle) {
    this.libelle = libelle;
  }

  List<Ressource> getRessource() {
    return ressources;
  }

  void addRessource(Ressource ressource) {
    if (!ressources.contains(ressource)) {
      ressources.add(ressource);
    }
  }

  void removeRessource(Ressource ressource) {
    ressources.remove(ressource);
  }

  TypeRelation.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    libelle = json['libelle'];
  }
}
